src/MonoTorrent/MonoTorrent.Client/Managers/InactivePeerManager.cs

```diff
@@ -61,8 +62,12 @@ internal void TimePassed ()
             int candidateSecondsConnected ;
             int candidateSecondsSinceLastBlock;
             int candidateDataBytes;
-            for (int i = 0; i < TorrentManager.Peers.ConnectedPeers.Count; i++) {
-                PeerId nextPeer = TorrentManager.Peers.ConnectedPeers[i];
+            PeerId[] arr = TorrentManager.Peers.ConnectedPeers.ToArray();
```

Comment:
what's the point in converting to array or to list here and below?

Reply:
if function CleanupSocket() changed the ConnectedPeers, exception argument out of range might get throwed. i'm not sure this is a good fix or not

Reply:
CleanupSocket is guaranteed to run on the `ClientEngine.MainLoop` thread. This code is also guaranteed to run on that thread. As such it should be impossible for CleanupSocket to run while this code is running, unless this code calls CleanupSocket.

I think we can undo this part of the patch based on that. Right?